apps/webcert/src/hooks/useGoBackEffect.ts

```diff
@@ -0,0 +1,22 @@
+import { useHistory } from 'react-router'
+import { useAppSelector } from '../store/store'
+import { getPatientId } from '../store/srs/srsSelectors'
+
+export const useGoBackEffect = () => {
+  const history = useHistory()
+  const patientId = useAppSelector((state) => getPatientId(state))
+  const isAfterSithsSign = document.referrer.startsWith('https://esign')
```

Comment:
Instead of doing document.location.reload() we could perhaps put this in a state that we "consume"

```
  const [isForreignReferrer, updateForreignReferrer] = useState(document.referrer.startsWith('https://esign'))

  return () => {
    if (isForreignReferrer) {
      if (patientId) {
        history.push(`/create/${encodeURIComponent(btoa(patientId))}`)
      } else {
        history.push('/search')
      }
      updateForreignReferrer(false)
    } else {
      history.goBack()
    }
  }
```

If this doesn't work, I would prefer if we use `window.location.replace` instead of `history.push` and remove `document.location.reload`.

Reply:
If we also check referrer against `window.location.hostname` we could perhaps make this a bit more general, supporting whatever domain may be in `document.referrer`